### PR TITLE
feat: add `StdSecp256k1` library

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
- add `A` / `B` / `P` / `N` / `GX` / `GY` from `Secp256k1` (`P`, `N`, `GX`, `GY` match `go-ethereum`)
- provide `(x,y) `/ `(x,y,z)` helpers: `identityAffine`, `isIdentityAffine`, `identityProjective`, `isIdentityProjective`, `generatorAffine`, `generatorProjective`
- keep `Vm` minimal (only conversions), pure helpers stay in `forge-std`

### CODE REVIEW

File: src/StdSecp256k1.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+/// - https://en.bitcoin.it/wiki/Secp256k1
+/// - https://github.com/ethereum/go-ethereum/blob/v1.17.5/crypto/secp256k1/curve.go#L267
+library StdSecp256k1 {
+    /// @dev Curve parameter `a = 0`.
+    uint256 internal constant A = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    /// @dev Curve parameter `b = 7`.
+    uint256 internal constant B = 0x0000000000000000000000000000000000000000000000000000000000000007;
+    /// @dev Prime number, public key `(x, y)`, where `(x, y)` must be in `[0, StdSecp256k1.P)`.
+    uint256 internal constant P = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFC2F;
+    /// @dev Prime number, scalar `s` must be in `[0, StdSecp256k1.N)`, non-zero scalar must be in `[1, StdSecp256k1.N)`.
```
Comment (mablr) [inline src/StdSecp256k1.sol]: `ecMul*` accepts any `uint256` and reduces it modulo `N`, so “must be in `[0, N)`” is misleading. Could we describe `N` as the group order and `[0, N)` as the canonical scalar range?

File: src/StdSecp256k1.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    function identityAffine() internal pure returns (uint256 x, uint256 y) {
+        return (0, 0);
+    }
+
+    /// @notice Returns the secp256k1 generator point `G` in affine coordinates as `(GX, GY)`.
+    function generatorAffine() internal pure returns (uint256 x, uint256 y) {
+        return (GX, GY);
+    }
+
+    /// @notice Returns whether the projective point `(x, y, z)` is the point at infinity.
```
Comment (mablr) [inline src/StdSecp256k1.sol:51]: Could we document that Foundry uses homogeneous coordinates `(X/Z, Y/Z)` here? “Projective” is otherwise easy to mistake for Jacobian `(X/Z², Y/Z³)`.

File: src/StdSecp256k1.sol [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    /// @dev `y` is ignored; infinity is `(0, y, 0)` for any non-zero `y` accepted by Vm,
+    /// but normalized form is `(0, 1, 0)`.
+    function isIdentityProjective(uint256 x, uint256, uint256 z) internal pure returns (bool) {
```
Comment (mablr) [inline src/StdSecp256k1.sol]: This also returns true for `(0, 0, 0)` and `(0, P, 0)`, which `Vm` rejects. Please include the accepted `y` range: `x == 0 && z == 0 && y != 0 && y < P`.

File: src/StdSecp256k1.sol [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+pragma solidity >=0.8.13 <0.9.0;
+
+/// @dev Library for interaction with secp256k1 elliptic curve,
+/// described by equation `y^2 = x^3 + ax + b (mod p)`
+/// where `a = 0` and `b = 7`.
+/// @dev Curve parameters taken from:
+/// - https://en.bitcoin.it/wiki/Secp256k1
+/// - https://github.com/ethereum/go-ethereum/blob/v1.17.5/crypto/secp256k1/curve.go#L267
+library StdSecp256k1 {
+    /// @dev Curve parameter `a = 0`.
```
Comment (mablr) [inline src/StdSecp256k1.sol:22]: Could we add focused tests for the constants, identity boundaries, constructors, and `Vm` interoperability—including a homogeneous point with `z != 1`?

File: src/StdSecp256k1.sol (hunk 1 above)
Comment (StackOverflowExcept1on) [reply src/StdSecp256k1.sol]: This is just docs about traditional secp256k1 scalars. Mb you want to use `addmod` / `mulmod` in EVM to do explicit scalar reduction, not implicit reduction.

Comment (mablr) [reply src/StdSecp256k1.sol]: Agreed. Maybe: “Group order. Scalars are canonically represented in [0, N); private keys in [1, N).”

**Review by mablr — COMMENTED** [review]
sgtm, pending review from a code owner

**Review by mattsse — APPROVED** [review]

**Comment by StackOverflowExcept1on** [comment]
Trigger CI plz, merged master

**Review by mablr — APPROVED** [review]

**Comment by mablr** [comment]
can't trigger here, @mattsse @grandizzy could u ?

**Comment by mattsse** [comment]
@StackOverflowExcept1on please dont modify the pr after it was approved

**Review by mattsse — APPROVED** [review]

**Comment by StackOverflowExcept1on** [comment]
Sorry, these are the requirements for auto-merge.
